source/CommandLine/ShellCompletion/CommandSuggester.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Octopus.CommandLine.ShellCompletion
+{
+    public static class CommandSuggester
+    {
+        public static IEnumerable<string> SuggestCommandsFor(
+            string[] words,
+            IReadOnlyDictionary<string, string[]> completionItems)
+        {
+            // some shells will pass the command name as invoked on the command line.
+            // If so, strip them from the beginning of the array
+            words = words
+                .Take(2)
+                //todo: MATTR: make this work for non-octo apps
+                .Except(new[] { "octo", "complete" }, StringComparer.OrdinalIgnoreCase)
```

Comment:
I see the todo here, but should this then be pulled across to the CommandLine repo if its still just for oco?

Reply:
Glad you caught that - it needed to be done.